pkg/scclient/client.go

```diff
@@ -954,12 +965,13 @@ func (c *RegistryClient) WatchMicroService(microServiceID string, callback func(
 }
 
 func (c *RegistryClient) getAddress() string {
-	next := RoundRobin(c.Config.Addresses)
-	addr, err := next()
-	if err != nil {
-		return DefaultAddr
+	for _, v := range c.Config.Addresses {
+		if c.Config.Status[v] == unavailable {
+			continue
+		}
+		return v
```

Comment:
after you filter ths address list ,you shoud use round robin to select one

Reply:
should I not return v, just create a temp slice store v, then use origin code pass it to RoundRobin()?

Reply:
sure

Reply:
done